parse.py

```diff
@@ -1,62 +1,80 @@
-                    Defaults to 'en'.
-    :return: Result in JSON format.
-    :param isTable: parses output as a table
-    """
-
-    payload = {'url': url,
-               'isOverlayRequired': overlay,
-               'apikey': api_key,
-               'language': language,
-               'istable': istable,
-               }
-    r = requests.post('https://api.ocr.space/parse/image',
-                      data=payload,
-                      )
-    return r.content.decode()
-
-
-print(ocr_space_file('receipt2.jpg'))
-print(ocr_space_url('https://ocr.space/Content/Images/receipt-ocr-original.jpg'))
+import requests, itertools
+import json
+from api import ocr_key
+
+
+def ocr_space_file(filename, overlay=False, api_key= ocr_key, language='eng', istable=True, scale = True):
+    """ OCR.space API request with local file.
+        Python3.5 - not tested on 2.7
+
+    :param filename: Your file path & name.
+    :param overlay: Is OCR.space overlay required in your response.
+                    Defaults to False.
+    :param api_key: OCR.space API key.
+                    Defaults to 'helloworld'.
+    :param language: Language code to be used in OCR.
+                    List of available language codes can be found on https://ocr.space/OCRAPI
+                    Defaults to 'en'.
+    :return: Result in JSON format.
+    :param istable: parses output as a table
+    """
+
+    payload = {'isOverlayRequired': overlay,
+               'apikey': api_key,
+               'language': language,
+               'istable': istable,
+               }
+    with open(filename, 'rb') as f:
+        r = requests.post('https://api.ocr.space/parse/image',
+                          files={filename: f},
+                          data=payload,
+                          )
+    return r.content.decode()
+
+
+def ocr_space_url(url, overlay=False, api_key= ocr_key, language='eng', isTable=True):
+    """ OCR.space API request with remote file.
+        Python3.5 - not tested on 2.7
+
+    :param url: Image url.
+    :param overlay: Is OCR.space overlay required in your response.
+                    Defaults to False.
+    :param api_key: OCR.space API key.
+                    Defaults to 'helloworld'.
+    :param language: Language code to be used in OCR.
+                    List of available language codes can be found on https://ocr.space/OCRAPI
+                    Defaults to 'en'.
+    :return: Result in JSON format.
+    :param isTable: parses output as a table
+    """
+
+    payload = {'url': url,
+               'isOverlayRequired': overlay,
+               'apikey': api_key,
+               'language': language,
+               'isTable': isTable,
+               }
+    r = requests.post('https://api.ocr.space/parse/image',
+                      data=payload,
+                      )
+    #return r.content.decode()
+    return r.json()
+
+data = ocr_space_url('https://ocr.space/Content/Images/receipt-ocr-original.jpg')
```

Comment:
Move this code to another file or guard it behind an `if __name__ == '__main__':` block to prevent it from being executed if this file is imported.

Reply:
Can you check the new commit to see if this addresses what you're suggesting? If not correct, please help with correct method either here or PM me

Reply:
Yep, looks good! Feel free to merge.